Develop

jackpr-unity opened the pull request:
Based off of StayTalm/InputSystem master branch, but PR-ing into develop because that's how this goes...?

jackpr-unity committed:
Start of XR test folder structure.  XR Devices scene prints out devices.

jackpr-unity committed:
Committing before upstream merge

jackpr-unity committed:
Merged with upstream

jackpr-unity committed:
committing before merge

jackpr-unity committed:
Merge remote-tracking branch 'upstream/master' into develop

jackpr-unity committed:
Adding new device-specific debug scenes.

Commit:
Update touch test to be compatible with Rene's latest commit b03c6f23b74b76596966a634a67ff74f354f8e17

jackpr-unity committed:
Merge remote-tracking branch 'upstream/master' into develop

jackpr-unity committed:
Removing unnecessary meta files

jackpr-unity committed:
- Added ViveWand test scene
- Added WMR visual asset

jackpr-unity committed:
Adding knuckles scene

StayTalm commented on Assets/QA/Tests/XRDeviceActions/_SharedScripts/AxisControlActionStatus.cs:
```diff
@@ -0,0 +1,35 @@
+
+public class AxisControlActionStatus : MonoBehaviour {
+
+	public InputAction axisAction;
```
don't forget to run Format.sh on this.

StayTalm commented on Assets/QA/Tests/XRDeviceActions/_SharedScripts/AxisControlActionStatus.cs:
```diff
@@ -0,0 +1,35 @@
+    {
+        float value = ((AxisControl)(context.control)).ReadValue();
+        statusSlider.value = value;
+        //statusText.text = value.ToString();
```
This should either be removed, or have a second comment explaining why it's temporarily removed (ideally with name/date).

StayTalm commented on Assets/QA/Tests/XRDeviceActions/_SharedScripts/AxisControlActionStatus.cs:
```diff
@@ -0,0 +1,35 @@
+        //statusText.text = value.ToString();
+    }
+
+    private void OnApplicationQuit()
```
This seems unnecessary.  If the application is quitting, do we still care about the action status?

Also if you want a proper enable/disable location, instead of using start, I'd suggest Monobehaviour.OnEnable and Monobehaviour.OnDisable.

StayTalm commented on Assets/QA/Tests/XRDeviceActions/_SharedScripts/Vector2ControlActionStatus.cs:
```diff
@@ -0,0 +1,40 @@
+        //status2Text.text = value.y.ToString();
+    }
+
+    private void OnApplicationQuit()
```
I keep seeing this Start/OnApplicationQuit pattern and it's incorrect.  You can use either
OnEnable/OnDisable
or
Start/OnDestroy
But Start and OnApplicationQuit are not parallel calls, if you destroyed a Monobehaviour manually it would cause some issue here.

StayTalm commented on Assets/QA/Tests/XRDeviceActions/_SharedScripts/Vector3ControlActionStatus.cs:
```diff
@@ -0,0 +1,41 @@
+        statusText.text = Vector3ToFieldText(value);
+    }
+
+    private string Vector3ToFieldText (Vector3 inVec)
```
I would suggest using the formatting version of this [https://docs.unity3d.com/ScriptReference/Vector3.ToString.html] instead.

jackpr-unity replied:
I chose to do it this way, and with a monospace font so that it is super easy to read.  Using the other method makes UI spacing difficult and can result in ugly and difficult to decipher results.

StayTalm replied:
Wouldn't this still be functionally equivalent to?
`inVec.ToString("+000.000;-000.000;+000.000");`
The Vector3 function is this:
```
public string ToString(string format)
        {
            return UnityString.Format("({0}, {1}, {2})", x.ToString(format), y.ToString(format), z.ToString(format));
        }
```

Duplicating the same behaviour in two places is not a good idea.

StayTalm commented on Assets/QA/Tests/XRDeviceActions/_SharedScripts/Vector2ControlActionStatus.cs:
```diff
@@ -0,0 +1,40 @@
+        Vector2 value = ((Vector2Control)(context.control)).ReadValue();
+        status1Slider.value = value.x;
+        status2Slider.value = value.y;
+        //status1Text.text = value.x.ToString();
```
You should either delete commented code or explain why it's still needed and can't be written once it's ready to actually use.

StayTalm commented on Assets/QA/Tests/XRDeviceActions/_SharedScripts/QuaternionControlActionStatus.cs:
```diff
@@ -0,0 +1,42 @@
+        statusText.text = QuaternionToFieldText(value);
+    }
+
+    private string QuaternionToFieldText(Quaternion inQuat)
```
Can you use https://docs.unity3d.com/ScriptReference/Quaternion.ToString.html to do the formatting?

jackpr-unity replied:
I chose to do it this way, and with a monospace font so that it is super easy to read. Using the other method makes UI spacing difficult and can result in ugly and difficult to decipher results.

StayTalm commented on Assets/QA/Tests/XRDeviceList/XRDeviceList.cs:
```diff
@@ -0,0 +1,37 @@
+        string deviceAccumulator = "";
+        for (int i = 0; i < InputSystem.devices.Count; i++)
+        {
+            deviceAccumulator += "<" + i + "> " + InputSystem.devices[i].displayName + "\n";
```
`deviceAccumulator += string.Format("<{0}> {1}\n", i, InputSystem.devices[i].displayName);`
Could be easier to parse and edit later

jackpr-unity replied:
tomato tomato

StayTalm commented on Assets/QA/Tests/XRHaptics/XRHaptics.cs:
```diff
@@ -0,0 +1,64 @@
+        rightHapticDetected.color = (rightHandController != null) ? Color.red : Color.white;
+        if (rightHandController != null)
+        {
+            //if (m_LeftHandRumbling)
```
This should be removed, as an unexplained commented code block.

StayTalm commented on Assets/QA/Tests/XRHaptics/XRHaptics.cs:
```diff
@@ -0,0 +1,64 @@
+        leftHapticDetected.color = (leftHandController != null) ? Color.red : Color.white;
+        if (leftHandController != null)
+        {
+            //if (!m_LeftHandRumbling)
```
This should be removed, as an unexplained commented code block.

StayTalm commented on Assets/QA/Tests/XRHaptics/XRHaptics.cs:
```diff
@@ -0,0 +1,64 @@
+    {
+        m_Timer += Time.deltaTime;
+
+        if (m_Timer >= m_RumblePeriod)
```
This probably not the right pattern since delta time is always somewhat over the period.  To sync it properly I suggest something like
```
m_Timer += Time.deltaTime;
if (m_Timer >= m_RumblePeriod)
{
    m_timer -= m_RumblePeriod;
    m_LeftHandRumbling = !m_LeftHandRumbling;
}

```

Which will stay consistent regardless of how delta time syncs up to the rumble period.

jackpr-unity replied:
All this does is swap rumble between left and right controllers once per period.  It doesn't have to be precise.

StayTalm reviewed (changes requested):
Some small suggestions and cleanup

jackpr-unity committed:
Cleanup based on suggestions from Tom for PR to his repository

jackpr-unity committed:
Changed formatting technique

jackpr-unity committed:
- Removed unused axis values from scenes
- fixed TouchscreenTouchVisualizer.cs DeleteExtraTouches
- Fixed events in ActionStatus scripts

jackpr-unity commented:
Please review newest changes. I believe the latest commit resolves the issues you have mentioned.

StayTalm commented:
Looks good now, going to dig into the merge conflict and merge :)

StayTalm committed:
Merge branch 'develop' into develop
